core/pkg/artifacts/saver.go

```diff
@@ -45,6 +52,7 @@ func NewArtifactSaver(
 		Artifact:            artifact,
 		HistoryStep:         historyStep,
 		StagingDir:          stagingDir,
+		batchSize:           1000,
```

Comment:
It probably doesn't have anything to do with the user's CPU count, but the ideal value would depend on average file size and network speed. Mostly the value doesn't really matter much, but if they take a long time (several minutes) to upload each file and have a lot of files (so that an artifact upload would take >24 hours) then it would be good for us to reduce the batch size to ensure we're not getting upload URLs far in advance of using them